src/lsdb/core/search/validators.py

```diff
@@ -0,0 +1,68 @@
+from __future__ import annotations
+
+from typing import List
+
+import numpy as np
+import numpy.typing as npt
+from hipscat.pixel_math.polygon_filter import SphericalCoordinates
+
+
+class CoordinatesValidator:
+    """Performs right ascension / declination coordinate validation"""
+
+    @staticmethod
+    def validate_cone_search_params(ra: float, dec: float, radius: float) -> float:
+        """Checks that the cone center has valid declination and wraps the value
+        for its right ascension. It also makes sure the cone radius is positive.
+
+        Arguments:
+            ra (float): The right ascension to the center of the cone
+            dec (float): The declination to the center of the cone
+            radius (float): The cone radius, in degrees
+
+        Returns:
+            The wrapped value for the right ascension of the cone center, in the
+            [0,360] degrees range.
+        """
+        if radius < 0:
+            raise ValueError("Cone radius must be non negative")
```

Comment:
Why is a radius of 0 a valid value?

Reply:
Good question, it's maybe a typo! @hombit @smcguire-cmu do you know if there's any reason for this?

Reply:
I changed it so now radii need to be positive. Please let me know if there's an issue with it!